wolfhsm/wh_server_cache.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2024 wolfSSL Inc.
+ *
+ * This file is part of wolfHSM.
+ *
+ * wolfHSM is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfHSM is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with wolfHSM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * wolfhsm/wh_server_cache.h
+ *
+ * Exists as a separate header so it can be consumed by server, server keystore, and NVM
+ * layer without creating circular dependencies
+ */
+
+#ifndef WOLFHSM_WH_SERVER_CACHE_H_
+#define WOLFHSM_WH_SERVER_CACHE_H_
+
+/* Pick up compile-time configuration */
+#include "wolfhsm/wh_settings.h"
+#include "wolfhsm/wh_common.h"
+
+#ifndef WOLFHSM_CFG_NO_CRYPTO
+
+/** Server cache slot structures */
+typedef struct whServerCacheSlot {
+    uint8_t       committed;
```

Comment:
```suggestion
    uint32_t       committed:1;
```
Consider allocating a bitfield here or a larger variable to avoid internal structures padding.

Reply:
I didn't change how it was defined before. Didn't want to go changing code outside the scope of this PR. Padding in this case doesn't matter, since it isnt a wire struct